tests/func/test_analytics.py

```diff
@@ -23,6 +26,40 @@ def test_main_analytics(mock_is_enabled, mock_report, tmp_dir, dvc):
     assert mock_report.called
 
 
+@pytest.fixture
+def mock_daemon(mocker):
+    def func(argv):
+        return main(["daemon", *argv])
+
+    m = mocker.patch("dvc.daemon.daemon", mocker.MagicMock(side_effect=func))
+    yield m
```

Comment:
Having a conditional daemonizer would be good for testing and QA, rather than mocking it. But didn't want to introduce big change.